cmd/server/helper_keys.go

```diff
@@ -18,7 +18,7 @@ func createRS256KeysIfNotExist(c *config.Config, set, kid, use string) {
 	if _, err := ctx.KeyManager.GetKey(set, kid); errors.Cause(err) == pkg.ErrNotFound {
 		logrus.Infof("Key pair for signing %s is missing. Creating new one.", set)
 
-		keys, err := generator.Generate("")
+		keys, err := generator.Generate(set)
```

Comment:
The issue with that is when you return all the public keys in the `well-known` response, they would all have the `kid` of `public`, and the breaks a lot of client libraries that are expecting a unique `kid` there, which is matched to the `kid` in the jwt header. So the `kid` needs to be more enriched than just `public` but perhaps we can still reference it that way with a layer of abstraction